frontend/src/metabase-lib/lib/Dimension.ts

```diff
@@ -815,80 +818,127 @@ export class FieldDimension extends Dimension {
     return field;
   }
 
-  field(): Field {
+  _getIdentifierProp() {
+    return this.isIntegerFieldId() ? "id" : "name";
+  }
+
+  _getTrustedFieldCachedOnInstance() {
     if (
       this._fieldInstance &&
       this._fieldInstance._comesFromEndpoint === true
     ) {
       return this._fieldInstance;
     }
+  }
 
-    const question = this.query()?.question();
-    const lookupField = this.isIntegerFieldId() ? "id" : "name";
-    const fieldMetadata = question
-      ? _.findWhere(question.getResultMetadata(), {
-          [lookupField]: this.fieldIdOrName(),
-        })
-      : null;
+  _getFieldMetadataFromSavedDataset() {
+    const identifierProp = this._getIdentifierProp();
+    const questionAssociatedWithDimension = this.query()?.question();
+    if (
+      questionAssociatedWithDimension?.isSaved() &&
+      questionAssociatedWithDimension.isDataset()
+    ) {
+      const question = this.query()?.question();
+      const field = _.findWhere(question.getResultMetadata(), {
+        [identifierProp]: this.fieldIdOrName(),
+      });
 
-    // Field result metadata can be overwritten for models,
-    // so we need to merge regular field object with the model overwrites
-    const shouldMergeFieldResultMetadata = question?.isDataset();
+      return field;
```

Comment:
based on the function name and where it's used (line 915-917) should it be renamed to `fieldMetadata`?

Reply:
Updated